src/components/feedback/Snackbar/index.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import styles from './styles.less';
+import classNames from 'classnames';
+
+export enum verticalPositions {
+    top = 'top',
+    bottom = 'bottom',
+}
+export enum horizontalPositions {
+    left = 'left',
+    right = 'right',
+    center = 'center',
+}
+
+type SnackBarProps = {
+    verticalPosition?: verticalPositions;
+    horizontalPosition?: horizontalPositions;
+    open?: boolean;
+    autoHideDuration?: number;
+    onClose?: () => void;
+    message?: string;
+    actions?: Array<JSX.Element>;
+};
+
+const SnackBar: React.FC<SnackBarProps> = ({
+    verticalPosition = verticalPositions.bottom,
+    horizontalPosition = horizontalPositions.left,
+    open,
+    autoHideDuration = 6000,
```

Comment:
Instead of the snackbar component handeling the "auto hide" it should be handled by the notification center. It could however accept an abort signal from an [AbortController ](https://developer.mozilla.org/en-US/docs/Web/API/AbortController)and hide when the signal is aborted.